### PR TITLE
Feature Layer graphics

### LINKED ISSUE #92: Support Graphics added to FeatureLayer
According to the [Feature Layer](https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-FeatureLayer.html) JS docs, a `Feature` in ArcGIS JS is just a `Graphic`.

- Add the ability to add and register Graphics as subcomponents of `FeatureLayer`
- Alter add, remove, and update logic for graphics to make sure it handles feature layer graphics as well as `GraphicLayer` graphics and those placed directly on `MapView`.

### PR DESCRIPTION
**Pull request by TimPurdum**
Closes #92 

- Added the ability to use `Graphics` as local, client-side features in a `FeatureLayer`
- Added `Field` for `FeatureLayer` attributes
- Added parameterized constructors for the following classes to avoid BL0005 warning (setting razor Parameters outside razor)
   - Extent
   - Point
   - PolyLine
   - Polygon
   - SpatialReference
   - Graphic
   - FieldInfo
   - FieldInfoFormat
   - FieldsPopupContent
   - PopupTemplate
   - PopupOptions
- Added convenience `Add` and `Remove` methods to `FeatureLayer`, `GraphicsLayer`, and `MapComponent` that call `RegisterChildComponent` (or `UnRegister...` for `Remove`) under the hood.
- Added to the `FeatureLayers.razor` sample page with an example of setting a feature layer from local graphics added in C#

### CODE REVIEW

**Comment by TimPurdum** [comment]
@morehavoc can you review this PR today?

File: samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/BasemapProjections.razor [hunk 1]
```diff
@@ -41,5 +41,5 @@
         return Task.CompletedTask;
     }
 
-    private SpatialReference _spatialReference = new() { Wkid = 102100 };
+    private SpatialReference _spatialReference = new(102100);
```
Comment (morehavoc) [inline samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/BasemapProjections.razor:44]: Does us changing these to have constructors constitute a breaking change?

File: src/dymaptic.GeoBlazor.Core/Components/Geometries/Geometry.cs [hunk 2]
```diff
@@ -11,20 +11,6 @@ namespace dymaptic.GeoBlazor.Core.Components.Geometries;
 [JsonConverter(typeof(GeometryConverter))]
 public class Geometry : MapComponent
 {
-    /// <summary>
```
Comment (morehavoc) [inline src/dymaptic.GeoBlazor.Core/Components/Geometries/Geometry.cs:14]: How come these are going away?

File: src/dymaptic.GeoBlazor.Core/Scripts/arcGisJsInterop.ts [hunk 3]
```diff
@@ -160,7 +158,7 @@ export async function buildMapView(id: string, dotNetReference: any, long: numbe
                 });
                 let center;
                 let spatialRef;
-                if (spatialReference !== undefined && spatialReference !== null) {
+                if (hasValue(spatialReference)) {
```
Comment (morehavoc) [inline src/dymaptic.GeoBlazor.Core/Scripts/arcGisJsInterop.ts:161]: Nice!

**Review by morehavoc — APPROVED** [review]

**Comment by codergrl** [comment]
@TimPurdum I got the chance to test the sample and then got pulled away on client calls, but while testing I noticed a bunch of errors you should probably look into before committing this. I basically just clicked to turn the layers on and off randomly in the sample. 

image

File: samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/BasemapProjections.razor (hunk 1 above)
Comment (TimPurdum) [reply samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/BasemapProjections.razor:44]: No, you can still call the parameterless constructor, but you will get a warning (which is how it was previously).

File: src/dymaptic.GeoBlazor.Core/Components/Geometries/Geometry.cs (hunk 2 above)
Comment (TimPurdum) [reply src/dymaptic.GeoBlazor.Core/Components/Geometries/Geometry.cs:14]: As we discussed on Discord, these were non-functional, so should not have been included. Setting them did nothing, and they would always return `null`. I will see if I can implement their real functionality and add them back in soon.

File: src/dymaptic.GeoBlazor.Core/Scripts/arcGisJsInterop.ts (hunk 3 above)
Comment (TimPurdum) [reply src/dymaptic.GeoBlazor.Core/Scripts/arcGisJsInterop.ts:161]: you can only write the same thing so many times.... 🙂

**Pull request merged**